library/TeslaAuthHelper.cs

```diff
@@ -167,137 +160,40 @@ public async Task<Tokens> RefreshTokenAsync(string refreshToken, CancellationTok
-
-            if (result.StatusCode != HttpStatusCode.Redirect && !result.IsSuccessStatusCode)
-            {
-                throw new Exception(string.IsNullOrEmpty(result.ReasonPhrase)
-                    ? result.StatusCode.ToString()
-                    : result.ReasonPhrase);
-            }
-
-            if (result.StatusCode != HttpStatusCode.Redirect)
-            {
-                if (result.StatusCode == HttpStatusCode.OK && resultContent.Contains("passcode"))
-                {
-                    if (string.IsNullOrEmpty(mfaCode))
-                    {
-                        throw new MultiFactorAuthenticationException(String.Format("Multi-factor code required to authenticate for account {0}", username), username);
-                    }
-
-                    return await GetAuthorizationCodeWithMfaAsync(mfaCode, loginInfo, client, cancellationToken);
-                }
-                else if (result.StatusCode == HttpStatusCode.OK)
-                {
-                    // CAPTCHA requirement, probably.
-                    bool hasCaptcha = resultContent.Contains("captcha");
-                    bool hasReCaptcha = resultContent.Contains("recaptcha");
-                    bool hasRedirectLocation = result.Headers.Location != null;
-                    if (!hasCaptcha)
-                    {
-                        // This is unexpected.  What is this?
-                        throw new Exception("Can't log in - expected redirect did not occur.  hasRedirect: " + hasRedirectLocation);
-                    }
-                    if (hasReCaptcha)
-                    {
-                        throw new Exception("Can't log in - Tesla ReCAPTCHA may be necessary");
-                    }
-                    throw new Exception("Can't log in - Tesla CAPTCHA support needed.");
-                }
-                else
-                {
-                    // Possible causes for ending up here:
-                    // 1) Account could require an MFA code and we didn't provide it, maybe.  Captcha check may come first though.
-                    // What happens with the wrong MFA code?  Does that come here?
-                    // We believe cases like the account being locked fall into a different codepath below.
-                    throw new Exception("Expected redirect did not occur - probably need multi-factor authentication code.  Status code: " + result.StatusCode);
-                }
-            }
-
-            var location = result.Headers.Location;
-
-            if (location == null)
-            {
-                throw new Exception("Redirect location not available");
-            }
-
-            string code = HttpUtility.ParseQueryString(location.Query).Get("code");
-            return code;
-        }
+        
 
         async Task<Tokens> ExchangeCodeForBearerTokenAsync(string code, HttpClient client, CancellationToken cancellationToken)
         {
             var body = new JObject

```

Comment:
Maybe use [System.Text.Json.Nodes](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.nodes?view=net-6.0) ? Could potentially remove the dependency on Newtonsoft.Json

Reply:
Thanks will do